design/one-pager-stack-relationship-labels.md

```diff
@@ -0,0 +1,49 @@
+# Stack Parent/Child Relationship Label Spec
+
+- Owner: Steven Rathbauer (@rathpc)
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#934](https://github.com/crossplaneio/crossplane/issues/934)
+
+## Problem
+
+There is currently no easy way to get a list of all resources that are owned by a parent resource. We can leverage CRD labels to make this easier. As long as we have a standard format for doing so.
+
+Per the proposal, it seems that the clearest example of what I am referring to is shown on this page https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels within the `app.kubernetes.io/part-of` label.
+
+As long as we define this label on all children CRD's, we should then be able to query all children of a given parent name using those labels.
```

Comment:
What are your thoughts on how to record multiple levels of parent/child relationships?

Reply:
What about differentiating resources of different instances of a particular CRD? `app.kubernetes.io/instance` looks like a good candidate on the page you linked.

Reply:
@turkenh yes thank you!

@suskin that is a good question. As of right now I was constraining this relationship pattern to only retrieving direct children of a parent.

It seems that down the line we can explore retrieving all children of a parent and children of those children. Additionally maybe even the other way around and getting parents from a given child. Although all of this obviously introduces far more complexity.

Is it fair to limit the scope of these relationships at this stage to just single levels?

Reply:
Hmm well, as I'm thinking more about this, I am realizing I may not have been thinking about the same use-cases as what the document is describing. @rathpc would you be able to also explain the specific use-cases we **may** want to satisfy (e.g. "get me all the children of my Wordpress object", "get me all the children of the Wordpress stack"), and also call out which ones we're trying to satisfy with this proposal?

Reply:
You hit the nail on the head there, the use case I am trying to satisfy with this proposal is "Get me all children of some parent object." Whether that is a stack or not shouldn't matter since the larger goal of this is to just be able to "dive" down through resources.

The most analogous example would be `node_modules` in npm. Some package in npm has dependencies and those dependencies may have their own dependencies and so on.

The same should be illustrated by this proposal where some stack has children crds and those crds can have other children crds and so on, while limiting the diving to one level at a time.

Does that help explain my thought process at all? @suskin